setup.py

```diff
@@ -3,44 +3,54 @@
 from __future__ import absolute_import
 
 import io
+import os
 import re
+import sys
+import codecs
 from setuptools import setup
+from setuptools.command.test import test as TestCommand
 
 
-def read(*filenames, **kwargs):
-    encoding = kwargs.get('encoding', 'utf-8')
-    sep = kwargs.get('sep', '\n')
-    buf = []
-    for filename in filenames:
-        with io.open(filename, encoding=encoding) as f:
-            buf.append(f.read())
-    return sep.join(buf)
+rootpath = os.path.abspath(os.path.dirname(__file__))
 
 
-VERSIONFILE = "gsw/__init__.py"
-verstrline = open(VERSIONFILE, "rt").read()
-VSRE = r"^__version__ = ['\"]([^'\"]*)['\"]"
-mo = re.search(VSRE, verstrline, re.M)
-if mo:
-    verstr = mo.group(1)
-else:
-    raise RuntimeError("Unable to find version string in %s." % (VERSIONFILE,))
+class PyTest(TestCommand):
+    def finalize_options(self):
+        TestCommand.finalize_options(self)
+        self.test_args = ['--strict', '--verbose', '--tb=long', 'tests']
+        self.test_suite = True
+
+    def run_tests(self):
+        import pytest
+        errno = pytest.main(self.test_args)
+        sys.exit(errno)
+
+
+def read(*parts):
+    return codecs.open(os.path.join(rootpath, *parts), 'r').read()
```

Comment:
I thought we might need to deal with encoding, but codecs without an encoding argument is equal to `open`.